### DISCUSSION TITLE
WordPress admin bar

### DISCUSSION THREAD

**Discussion by FabioCodex** [swup/swup#713] [Help] [answered]
Salve a tutti,
Avrei bisogno del vostro aiuto: sto creando un sito su wp in cui voglio usare swup per le transizioni. Funziona tutto correttamente, eccetto un problema: quando sono sul backend e vado da una pagina es Home e poi in Chi siamo e da qui clicco su 'Modifica pagina, l editor gitenberg mi porta alla pagina edit della Home e non i 'Chi siamo'...ho notato che se invece mi trovo su 'Chi siamo', faccio refresh e vado su modifica, l edit wp mi porta sulla modifica della pagina corrente ovvero 'Chi siamo', ma se mi sposto su un altra pagina e clicco su modifica pagina, mi porta sulla pagina in cui ho fatto il refresh all inizio.
Qualcuno di voi ha mai dovuto affrontare in passato tale problema e in caso sapere come ha risolto?
Grazie in anticipo.

**Comment by hirasso** [comment]
Hello @FabioCodex , l'm afraid none of us speaks Italian 😅

Could you translate your question to English? Thank you 🙏

**Comment by hirasso** [comment, marked as answer]
...translated your question myself. Make sure you either

- include your edit link inside of the `#swup` container
- add the edit link itself to swup's `containers` option.

**Reply by FabioCodex** [reply]
Hi,
Sorry for my misunderstanding.
So: in my linkseleector i added the class ab-item :no([data-no-swup)]...ab-item is a wp_admin_bar's link for 'Edit Page', but nothing change...I noticed , with opened inspector, that when I click the link of another page in nav menu, in frontend the page changes with correct transition, but seeing inspector in the same moment, the Url of the current page in Ab-item doesnt change.
Im working in localhost at the moment, so my site is not online yet

**Reply by daun** [reply]
@FabioCodex Have you tried adding the container of the admin bar to swup's `containers` option? The link selector shouldn't have anything to do with it.

**Reply by FabioCodex** [reply]
Hello daun,
yes I did....I adedd containers: ['#swup', '#wpadminbar']
But this solution retrieves an error: when I click in different page and click Edit Page seems that wp backend is corrupted, not displaying a page. My swup container is located just after my Header in header.php and it closing in footer.php just before the tag footer. This because i wanna dipslay both header and foter during transiction and changing only the content of the pages.
In short: the problem is admin bar that seems not included in swup container, I think.

**Reply by FabioCodex** [reply]
I trying everything in orde to solve this but I can't

**Reply by daun** [reply]
This is hard to debug with seeing the actual markup of the page. Any chance you could make it public or create a reproducible demo?

**Reply by FabioCodex** [reply]
I can show you a exampe of my template's code:

This is my Header.php
<html <?php language_attributes(); ?>>
<head>
    <meta charset="<?php bloginfo( 'charset'); ?>">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta description="<?php bloginfo('description'); ?>">
   
    <?php wp_head(); ?>
</head>
<?php wp_body_open(); ?>
<body <?php body_class(); ?>>
  
 

    <header>
    <nav class="navbar navbar-expand-lg navbar-light bg-transparent">
    <div class="container">
    <a class="navbar-brand" href="<?php echo esc_url(home_url()); ?>" id="positivo"><img src="<?php echo get_template_directory_uri(); ?>/img/hobos_logo.png" width="67" height="84" alt="hobos"></a>
          <a class="navbar-brand" href="<?php echo esc_url(home_url()); ?>" id="negativo"><img src="<?php echo get_template_directory_uri(); ?>/img/hobos_logo_negativo.png" width="67" height="84" alt="hobos"></a>
        <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#main-menu" aria-controls="main-menu" aria-expanded="false" aria-label="Toggle navigation">
            <span class="navbar-toggler-icon"></span>
        </button>
        
        <div class="collapse navbar-collapse" id="main-menu">
            <?php
            wp_nav_menu(array(
              'menu' => 'header',
                'theme_location' => 'header',
                'container' => 'div',
                'menu_class' => 'navbar-nav ms-auto mb-2 mb-lg-0',
                'items_wrap' => '<ul id="%1$s" class="navbar-nav ms-auto mb-2 mb-lg-0      %2$s">%3$s</ul>',
                'fallback_cb' => '__return_false',
                 'depth' => 2,
                'walker' => new bootstrap_5_wp_nav_menu_walker()
            ));
            ?>
            <a href=""><i class="fa-brands fa-facebook me-3"></i></a>
            <a href=""><i class="fa-brands fa-instagram"></i></a>
        </div>
    </div>
</nav>
</header>

<div id="swup" class="transition-fade">

here starts my swup container

**Reply by FabioCodex** [reply]
here ther's script init of my const Swup:

const options = {
  animateHistoryBrowsing: true,
  animationSelector: '[class*="transition-"]',
  containers: ["#swup"],
  cache: false,
    preload: true,
         linkSelector:
    'a[href^="' + window.location.origin +
    '"]:not([data-no-swup]):not([class*="ab-item"]), a[href^="/"]:not([data-no-swup]), a[href^="#"]:not([data-no-swup])',
    plugins: [ new SwupBodyClassPlugin(),
      new SwupHeadPlugin()
    ],
    skipPopStateHandling: function(event) {
    if (event.state && event.state.source == "swup") {
      return false;
    }
    
    return true;
  }


I have a Supreplacecontent event too in where there are other script that I reinitialized, and these ones work correctly after every transition

**Reply by FabioCodex** [reply]
If my site is offline, how can I make a demo of a site in offline? In order to show the issue that I explained?

**Reply by daun** [reply]
@FabioCodex  You could put it on a subdomain of the project's server, e.g. `test.example.com`.

**Reply by FabioCodex** [reply]
Hello guys! I'm here again with another swup issue....
Someone of you have ever combined Gsap and swup together?.....I ve imported GSAP for ScrollSmoother and after every transition, I notice that the next page, is more longer than I projected. Sometimes, the transition duplicate the footer section per page.
I trying to find  out on the net some documentation about it.

**Reply by daun** [reply]
@FabioCodex There's an example of GSAP animations in the [JS Plugin readme](https://swup.js.org/plugins/js-plugin/#gsap), that should give you some pointers.

**Comment by FabioCodex** [comment]
thanks a lot Dun...but the problem is always there....I don't know the why.
Swup is initialized just I wrote you in the old comments. Menawhile I declared ScrollSmooter like this-->
let smoother= ScrollSmoother.create({
   wrapper: "#smooth_wrapper",
   content: "#smooth_content",
   smooth: 1,
   smoothTouch: 0.1
 }).
In which smooth-wrapper is the main wrapper div of the page, and inside there's the smooth -content div, hust specified in gsap documentation...........So, I'm in stuck!

**Reply by daun** [reply]
I'd suggest you create a [minimal reproduction](https://swup.js.org/other/minimal-reproductions/) of the problem and post it in a new discussion, along with a clear description of the issue and what you've tried so far. It's impossible for others to help you without the whole context, including markup, styles and scripts. Creating a minimal reproduction will also help you narrow down the problem much faster.

**Comment by sanjaysubba** [comment]
The simple way that I've done is

```
  const swup = new Swup({
    containers: document.getElementById('wpadminbar') ? ['#main', '#wpadminbar'] : ['#main'],
    plugins: [new SwupHeadPlugin(), new SwupPreloadPlugin()],
    animateHistoryBrowsing: true,
    ignoreVisit: (url, { el }) => {
      //eslint-disable
      return el?.closest('#wpadminbar') || el?.hasAttribute('data-no-swup') || el?.closest('[data-no-swup]');
    },
  });
```

here "#main" is my main container
